### PR TITLE
misc: allow custom -userdatapath

### LINKED ISSUE #10: Hardcoded USERDATA_PATH
Hi team!

It looks like the `-userdatapath` argument is hardcoded to `/sonsoftheforest/userdata`. This makes it difficult for folks that want to pass the configuration files into the container, as that path is nested under the `GAME_PATH`.

Could we make `USERDATA_PATH` default to `/sonsoftheforest/userdata` but allow overrides from the environment?

### PR DESCRIPTION
**Pull request by justmiles**
Allows setting a custom `-userdatapath`. 

closes #10

### CODE REVIEW

**Review by jammsen — CHANGES_REQUESTED** [review]

File: usr/bin/servermanager.sh [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 # SteamCMD APPID for sons-of-the-forest-dedicated-server
 GAME_PATH="/sonsoftheforest/"
-USERDATA_PATH="/sonsoftheforest/userdata/"
-CONFIGFILE_PATH="/sonsoftheforest/userdata/dedicatedserver.cfg"
+USERDATA_PATH="${USERDATA_PATH:-/sonsoftheforest/userdata}"
```
[the remark this answers is not in the document]
Comment (justmiles) [reply usr/bin/servermanager.sh:5]: This is bash shorthand to set a default variable if unset. We could certainty move the default to the Dockerfile if that's preferred.

It's definitely valid, though. Here's a quick example using an online compiler: http://tpcg.io/_PE7JLT

File: usr/bin/servermanager.sh [hunk 2]
```diff
@@ -60,10 +60,15 @@ function installServer() {
     isWineinBashRcExistent
     steamcmdinstaller.sh
     mkdir -p $USERDATA_PATH
-    cp /dedicatedserver.cfg.example $CONFIGFILE_PATH
-    cp /ownerswhitelist.txt.example $USERDATA_PATH/ownerswhitelist.txt
+
+    # only copy examples if no config file exists
+    if [ ! -f $CONFIGFILE_PATH ]; then
```
[the remark this answers is not in the document]
Comment (justmiles) [reply usr/bin/servermanager.sh:65]: Makes sense, I'll add the `ownerswhitelist.txt` from your second example

File: usr/bin/servermanager.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (jammsen) [reply usr/bin/servermanager.sh:5]: Ahhh now i understand that line, i did never hear of this before, thats a lot for teaching me that!

But to be honest, i would prefer it the clean way, set all defaults in the Dockerfile and then you can overwrite it anyway you like after this.

[the remark this answers is not in the document]
Comment (jammsen) [reply usr/bin/servermanager.sh:5]: Do you want to do this or should i do this?

[the remark this answers is not in the document]
Comment (justmiles) [reply usr/bin/servermanager.sh:5]: Happy to update it. I'll knock it out shortly.

**Comment by justmiles** [comment]
@jammsen this is ready for review.

**Comment by jammsen** [comment]
Thanks @justmiles i didnt have time until yet, im gonna check this out now.

**Comment by jammsen** [comment]
Hey @justmiles 
So aside from a windows-vs-unix encoding problem which can easily be solved by "dos2unix servermanager.sh" and missing chmod +x permissions which might be down to my setup, everything else is working just fine.

Im done pushing the image right now to Docker-Hub - Feel free to test it and give me a feedback if what you want to do works 😃

**Pull request merged**